app/src/main/java/org/mozilla/tv/firefox/MainActivity.kt

```diff
@@ -121,12 +84,32 @@ class MainActivity : LocaleAwareAppCompatActivity(), OnUrlEnteredListener, Media
         }
     }
 
-    override fun onNewIntent(unsafeIntent: Intent) {
-        IntentValidator.validate(this, unsafeIntent.toSafeIntent(), ::onValidBrowserIntent)
+    @SuppressLint("MissingSuperCall")
+    override fun onSaveInstanceState(outState: Bundle?) {
+        // Do not save instance state.
```

Comment:
Overriding this and not calling super so state is not restored feels like a code smell: we don't know what the system is doing and it's not documented that we can override this to remove this behavior so we could be unknowingly breaking some functionality.

I think it'd be more appropriate to:
- clear the fragment back stack before recreation
- On restoration, if we're in the process of clearing data, modify the back stack
- ?

Reply:
The documentation does say that we can do this, as long as we're prepared to save all our application state:
```
The default implementation takes care of most of the UI per-instance state for you by calling View.onSaveInstanceState() on each view in the hierarchy that has an id, and by saving the id of the currently focused view (all of which is restored by the default implementation of onRestoreInstanceState(Bundle)). If you override this method to save additional information not captured by each individual view, you will likely want to call through to the default implementation, **otherwise be prepared to save all of the state of each view yourself**. 
```
https://developer.android.com/reference/android/app/Activity.html#onSaveInstanceState(android.os.Bundle)

Reply:
Also, since we don't have screen rotations, what are our configuration changes?  Clear data, change language, initial startup, is there anything else?  In the scenarios that I can think of, I think the only data we want to keep is stuff that's already being held in repos.  Things like focus and active fragment all _should_ be lost.  Personally, I see this as getting rid of some black magic that doesn't really apply to our situation.

Reply:
> The documentation does say that we can do this,

So it does! I looked at the wrong documentation.

> Also, since we don't have screen rotations, what are our configuration changes?

Save/restore logic also applies to memory eviction: for example, if the user moves from FF to Netflix and back, save/restore can be used to restore the user's website and back stack which we don't store in the repos, afaik. In my opinion, **we want to be restoring their state**: users don't expect to randomly lose their session due to an implementation detail.

Reply:
We currently restore the back/forward history if the WV is destroyed, as well as video position and where the user is scrolled to on the page, but I do take your point.

Reply:
Added this to handle in #1644.